gdnative-core/src/geom/basis.rs

```diff
@@ -164,4 +550,80 @@ mod tests {
         assert!(basis.elements[1] == Vector3::new(4.0, 5.0, 6.0));
         assert!(basis.elements[2] == Vector3::new(7.0, 8.0, 9.0));
     }
+
+    #[test]
+    fn basis_functions() {
```

Comment:
It's best to have separate `#[test]` functions for each of the methods, since it's easier to see which one failed from the output of `cargo test` that way. For the input vectors, you can introduce a helper function in the `test` module, for example:

```rust
fn test_inputs() -> (Vector3, Vector3, Basis, Basis) {
    let v = Vector3::new(37.51756, 20.39467, 49.96816);
    let vn = v.normalize();
    let b = Basis::from_euler(v);
    let bn = Basis::from_euler(vn);
    (v, vn, b, bn)
}

#[test]
fn scaled_is_sane() {
    let (v, vn, b, bn) = test_inputs();

    let expected = Basis::from_elements([
        Vector3::new(0.052484, -0.111876, 0.666453),
        Vector3::new(0.012407, -0.040492, -0.007774),
        Vector3::new(-0.682131, -0.212475, 0.018051),
    ]);
    assert!(expected.approx_eq(&b.scaled(&Vector3::new(0.677813, -0.043058, 0.714685))));
}

// -snip-
```